packages/blade/src/components/Form/FormLabel.tsx

```diff
@@ -16,7 +16,7 @@ type CommonProps = {
 };
 
 type LabelProps = CommonProps & {
-  htmlFor: string;
+  htmlFor?: string;
```

Comment:
Since I'm adding `FormLabel` outside of the `BaseInput` instead of using `BaseInput`'s `label` prop, I can't be certain on what the generated id would be for the first `BaseInput` and hence can't pass it as `htmlFor`.

> I can't pass `label` directly to `BaseInput` since it will move the first label down vertically causing it to break alignment with rest of the labels and the space for the label would be limited to that of the first input's width if we attach the label with the first `BaseInput`

Reply:
Ideally, there should be another hidden input which will act as a proxy for the whole input. And the segmented OTP inputs will just be for UX/show. 

Basically

```ts
       <FormLabel htmlFor="otpinput-1" as="label" position={labelPosition}>
          {label}
        </FormLabel>
        <Box display="flex" flexDirection="row">
          {inputs}
        </Box>
        <HiddenInput id="otpinput-1" value={otpValue.join('')} />
```

Reply:
Added a hidden input